src/scripts/storage/clipperData.ts

```diff
@@ -0,0 +1,49 @@
+import {GetResponseAsync, TimeStampedData} from "../http/cachedHttp";
+import {ClipperCachedHttp} from "../http/clipperCachedHttp";
+
+import {Logger} from "../logging/logger";
+
+import {ClipperStorageGateStrategy} from "./clipperStorageGateStrategy";
+import {Storage} from "./storage";
+import {StorageGateStrategy} from "./StorageGateStrategy";
+
+/**
+ * The data-access-like object that handles the fetching and caching of data
+ * from Clipper-specific endpoints (TODO: a factory class that is responsible for
+ * providing getRemoteValue objects?). Also provides a gating mechanism as a
+ * sanity-check for what information gets stored locally.
+ */
+export class ClipperData implements Storage {
+	private storage: Storage;
+	private storageGateStrategy: StorageGateStrategy;
+	private cachedHttp: ClipperCachedHttp;
+
+	constructor(storage: Storage, logger?: Logger) {
+		this.storage = storage;
+		this.storageGateStrategy = new ClipperStorageGateStrategy(storage);
+		// We pass 'this' as the Storage object as it handles all the sanity-check gating in the setValue
+		this.cachedHttp = new ClipperCachedHttp(this, logger);
+	}
+
+	public setLogger(logger: Logger) {
+		this.cachedHttp.setLogger(logger);
+	}
+
+	public getAndCacheFreshValue(key: string, getRemoteValue: GetResponseAsync, updateInterval?: number): Promise<TimeStampedData> {
```

Comment:
might just be me, but saying get _and_ cache feels like this is now promising to make the network call (get) so that it can follow through on its new promise to cache, which isn't what's happening here. a get will always happen of some sort, but it's possible the caching will not if the value was already fresh.

did others find the old `getFreshValue` name confusing? perhaps it's familiarity, but I was fine with that